core/src/main/java/io/grpc/internal/AbstractClientStream.java

```diff
@@ -119,8 +124,14 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
     }
     if (headers.containsKey(ChannelImpl.MESSAGE_ENCODING_KEY)) {
       String messageEncoding = headers.get(ChannelImpl.MESSAGE_ENCODING_KEY);
-      setDecompressor(messageEncoding);
+      try {
+        setDecompressor(messageEncoding);
+      } catch (IllegalArgumentException e) {
+        inboundTransportError(Status.INVALID_ARGUMENT.withDescription(e.getMessage()));
```

Comment:
This looks like it should be INTERNAL, since the application didn't specify an invalid argument. Instead, either the grpc library on client or server has a bug.